api.planx.uk/modules/flows/updateTemplatedFlow/service.ts

```diff
@@ -95,3 +97,32 @@ export const updateTemplatedFlow = async (
     commentId: insertCommentResponse.comment.id,
   };
 };
+
+export const customMerge = (
+  sourceData: Flow["data"], // FlowGraph ??
+  edits: Flow["data"],
+): Flow["data"] => {
+  // Preserves edited empty arrays (eg if a templated folder is emptied)
+  const mergedData = mergeWith(sourceData, edits, function (sourceData, edits) {
+    if (isArray(sourceData)) {
+      if (edits) {
+        return edits;
+      } else {
+        return sourceData;
+      }
+    }
+  });
+
+  // Check for orphaned nodes IDS (deleted edges) and remove them to ensure graph displays correctly
+  const validEdges = Object.values(mergedData).flatMap(
+    (nodeData) => nodeData.edges,
+  );
+  const orphanedNodeIds = Object.keys(mergedData).filter(
+    (nodeId) => nodeId !== "_root" && !validEdges.includes(nodeId),
+  );
+  orphanedNodeIds.forEach((nodeId) => {
+    delete mergedData[nodeId];
+  });
```

Comment:
The `mergeWith` above correctly reconciles arrays which means that our _navigation_ store methods will work correctly. We need to do this additional step of ensuring the overall flow graph doesn't have any now-orphaned nodes lingering to prevent them from displaying in the graph and "Search" or "Customise" sidebars.